src/main/scala/devices/uart/UART.scala

```diff
@@ -205,7 +204,7 @@ trait HasUARTTopModuleContents extends Module with HasUARTParameters with HasReg
   val rxm = Module(new UARTRx(params))
   val rxq = Module(new Queue(rxm.io.out.bits, uartNRxEntries))
 
-  val divinit = p(DTSTimebase) * BigInt(p(RTCPeriod).getOrElse(1)) / 115200
+  val divinit = p(PeripheryBusParams).frequency / 115200
```

Comment:
I think @mwachs5 is right that you should take this as a constructor arg and access p() in the trait. UART might not be on pbus.